paint/PAINT_DrawWindow.h

```diff
@@ -37,9 +38,11 @@ namespace paint
 
 		bool mouseButtonDown(win::MouseButton button) override;
 		bool mouseButtonUp(win::MouseButton b) override;
+		bool mouseExit(win::MouseButton button) override;
 		void setActiveTool(std::shared_ptr<Tool> tool);
 		std::shared_ptr<Tool> getActiveTool() const { return activeTool_; };
 		void toggleDrawTool(win::ToggleButton* b);
+		void toggleShapeTool(win::ToggleButton* b);
```

Comment:
No need to include ToogleButton (or Tool), just use a forward reference. Header files should always contain the minimum number of includes necessary for compilation.
**MUST**

Reply:
done